zcash_client_sqlite/src/lib.rs

```diff
@@ -1425,11 +1485,49 @@ impl<P: consensus::Parameters> WalletWrite for WalletDb<rusqlite::Connection, P>
                                 txout.value,
                                 None,
                             )?;
+
+                            // Even though we know the funding account, we don't know that we have
+                            // information for all of the transparent inputs to the transaction.
+                            #[cfg(feature = "transparent-inputs")]
+                            {
+                                tx_has_wallet_outputs = true;
+                            }
+
+                            // If the decrypted transaction is unmined and has no shielded
+                            // components, add it to the queue for status retrieval.
+                            #[cfg(feature = "transparent-inputs")]
+                            if d_tx.mined_height().is_none() && !detectable_via_scanning {
+                                wallet::queue_tx_retrieval(
+                                    wdb.conn.0,
+                                    std::iter::once(d_tx.tx().txid()),
+                                    TxQueryType::Status,
+                                    None
+                                )?;
+                            }
```

Comment:
This can't work I think. At this point an Enhancement request for the txid (the one we're processing) is already on the queue. The Status request will be ignored and then the call to `notify_tx_retrieved` on line 1529 will remove the Enhancement request from the queue.

Reply:
Good catch; I think that the `notify_tx_retrieved` call needs to be guarded by `detectable_via_scanning`.

Reply:
Unresolved so that I can link to this thread in another comment.